tests/migration_test.go

```diff
@@ -2866,53 +2866,30 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 				Expect(supportedFeatures).Should(Not(ContainElement(false)),
 					"copy features must be supported on node")
 			}
-			getOtherNodes := func(nodeList *k8sv1.NodeList, node *k8sv1.Node) (others []*k8sv1.Node) {
-				for _, curNode := range nodeList.Items {
-					if curNode.Name != node.Name {
-						others = append(others, &curNode)
-					}
-				}
-				return others
-			}
-			isHeterogeneousCluster := func() bool {
-				nodes := libnode.GetAllSchedulableNodes(virtClient)
-				for _, node := range nodes.Items {
-					hostModel := getNodeHostModel(&node)
-					otherNodes := getOtherNodes(nodes, &node)
-
-					foundSupportedNode := false
-					foundUnsupportedNode := false
-					for _, otherNode := range otherNodes {
-						if isModelSupportedOnNode(otherNode, hostModel) {
-							foundSupportedNode = true
-						} else {
-							foundUnsupportedNode = true
-						}
-
-						if foundSupportedNode && foundUnsupportedNode {
-							return true
-						}
-					}
-				}
-
-				return false
-			}
 
 			It("[test_id:6981]should migrate only to nodes supporting right cpu model", func() {
-				if !isHeterogeneousCluster() {
-					log.Log.Warning("all nodes have the same CPU model. Therefore the test is a happy-path since " +
-						"VMIs with default CPU can be migrated to every other node")
+				sourceNode, targetNode, err := getValidSourceNodeAndTargetNodeForHostModelMigration(virtClient)
+				if err != nil {
+					Skip(err.Error())
 				}
 
-				By("Creating a VMI with default CPU mode")
-				vmi := alpineVMIWithEvictionStrategy()
-
-				if cpu := vmi.Spec.Domain.CPU; cpu != nil && cpu.Model != v1.CPUModeHostModel {
-					log.Log.Warning("test is not expected to pass with CPU model other than host-model")
+				vmi := libvmi.NewAlpine(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					withEvictionStrategy(v1.EvictionStrategyLiveMigrate),
```

Comment:
Wouldn't it make sense to move the method into libvmi?

Reply:
Hey Daniel,
Thanks for reviewing :)
We had a discussion about it once:
https://github.com/kubevirt/kubevirt/pull/8075#discussion_r919053247